Common/Data/Custom/Intrinio/IntrinioEconomicData.cs

```diff
@@ -0,0 +1,231 @@
+
+
+        /// <summary>
+        ///     Return the URL string source of the file. This will be converted to a stream
+        /// </summary>
+        /// <param name="config">Configuration object</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>
+        ///     String URL of source file.
+        /// </returns>
+        /// <remarks>
+        ///     Given Intrinio's API limits, we cannot make more than one CSV request per second. That's why in backtesting mode
+        ///     we make sure we make just one call to retrieve all the data needed. Also, to avoid the problem of many sources
+        ///     asking the data at the beginning of the algorithm, a pause of a second is added.
+        /// </remarks>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            SubscriptionDataSource subscription;
+            var intrinioApiCallLimit = 1000 - (int)(DateTime.Now - LastIntrinoAPICall).TotalMilliseconds;
+
+            // We want to make just one call with all the data in backtesting mode.
+            // Also we want to make one call per second becasue of the API limit.
+            if (_firstTime || intrinioApiCallLimit > 0)
+            {
+                var order = string.Empty;
+                if (isLiveMode)
+                {
+                    // In Live mode, we only want the last observation.
+                    order = "desc";
+                }
+                else
+                {
+                    // In backtesting mode, we need the data in ascending order...
+                    order = "asc";
+                    // and avoid making furthers calls for the same data.
+                    _firstTime = false;
+                }
+
+                var item = GetStringForDataTransformation(_dataTransformation);
+                var url = string.Format("{0}identifier={1}&item={2}&sort_order={3}", _baseUrl,
+                                        config.Symbol.Value, item, order);
+                var byteKey = Encoding.ASCII.GetBytes(string.Format("{0}:{1}", _user, _password));
+                var authorizationHeaders = new List<KeyValuePair<string, string>>
+                {
+                    new KeyValuePair<string, string>("Authorization",
+                                                     string.Format("Basic ({0})", Convert.ToBase64String(byteKey)))
+                };
+
+                subscription = new SubscriptionDataSource(url, SubscriptionTransportMedium.RemoteFile, FileFormat.Csv,
+                                                          authorizationHeaders);
+                LastIntrinoAPICall = DateTime.Now;
+            }
+            else
+            {
+                subscription = new SubscriptionDataSource("", SubscriptionTransportMedium.LocalFile);
```

Comment:
Awesome. Empty source means do nothing. This is valid so long as we retrieve everything for backtest on the first invocation. What about multiple symbols?

Reply:
I tested it in the `BasicTemplateIntrinioEconomicData` algorithm and it works fine with two symbols.